eth/stagedsync/stage_tevm.go

```diff
@@ -45,22 +45,81 @@ func StageTranspileCfg(
 	}
 }
 
-func transpileBatch(logPrefix string, s *StageState, fromBlock uint64, toBlock uint64, tx ethdb.RwTx, batch ethdb.DbWithPendingMutations, cfg TranspileCfg, useExternalTx bool, quitCh <-chan struct{}) error {
-	logEvery := time.NewTicker(logInterval)
-	defer logEvery.Stop()
+func SpawnTranspileStage(s *StageState, tx ethdb.RwTx, toBlock uint64, quit <-chan struct{}, cfg TranspileCfg) error {
+	// panics! tx == nil
+
+	prevStageProgress, errStart := stages.GetStageProgress(tx, stages.Execution)
+	if errStart != nil {
+		return errStart
+	}
```

Comment:
yes, it's not very cool - but maybe can just do `cfg.db.View` if `tx == nil` here.